pkg/k8s/watchers/pod.go

```diff
@@ -584,6 +600,12 @@ func (k *K8sWatcher) upsertHostPortMapping(oldPod, newPod *slim_corev1.Pod, oldP
 	if !option.Config.EnableHostPort {
 		return nil
 	}
+
+	// no-op if specs are the same
```

Comment:
This should be moved to `updatePodHostData` before the defer function that does diff, and later `upsertHostPortMapping` is called. 1) The check applies to regular pods as well (not just for `hostPort` pods) 2) The diff is unnecessary if pod spec data isn't changed regardless.